spyder/widgets/editor.py

```diff
@@ -1718,12 +1723,11 @@ def current_changed(self, index):
                 pass
 
     def _get_previous_file_index(self):
-        if len(self.stack_history) > 1:
-            last = len(self.stack_history)-1
-            w_id = self.stack_history.pop(last)
-            self.stack_history.insert(0, w_id)

```

Comment:
Why did you remove this?

Reply:
This was an unnecessary side effect, that could cause the corruption of the historystack order.

This reorder the editor stack before closing the file, the closing could be cancelled, an the editor stack will have a wrong order.

Also this was unnecessary, because when a file is closed the CurrentChanged  signal is be emited, and that cause the history_stack to be updated.

Reply:
Ok, thanks for the explanation :-)